src/components/baseComponent.ts

```diff
@@ -95,6 +113,29 @@ export abstract class MgtBaseComponent extends LitElement {
     return !MgtBaseComponent._useShadowRoot || !(this.constructor as typeof MgtBaseComponent)._useShadowRoot;
   }
 
+  /**
+   * Invoked when the element is first updated. Implement to perform one time
+   * work on the element after update.
+   *
+   * Setting properties inside this method will trigger the element to update
+   * again after this update cycle completes.
+   *
+   * @param _changedProperties Map of changed properties with old values
+   */
+  protected firstUpdated(changedProperties): void {
+    super.firstUpdated(changedProperties);
+    Providers.onProviderUpdated(() => this.reload());
+    this.reload();
+  }
+
+  /**
+   * load state into the component.
+   * Override this function to provide additional loading logic.
+   */
+  protected load(): Promise<void> {
```

Comment:
I like this approach a lot. But I think we can be more descriptive since this is specific for loading the state. What about calling these methods `loadState` and `reloadState`

Reply:
Good idea. I like the clarity